buildSrc/src/main/java/com/uber/okbuck/generator/BuckFileGenerator.java

```diff
@@ -45,12 +47,19 @@ private BuckFileGenerator() {}
 
   /** generate {@code BUCKFile} */
   public static void generate(
-      Project project, BuckFileManager buckFileManager, VisibilityExtension visibilityExtension) {
-    List<Rule> rules = createRules(project);
-
+      Project project,
+      BuckFileManager buckFileManager,
+      VisibilityExtension visibilityExtension,
+      TestExtension testExtension) {
     File moduleDir = project.getBuildFile().getParentFile();
     File visibilityFile = new File(moduleDir, visibilityExtension.visibilityFileName);
     boolean hasVisibilityFile = visibilityFile.isFile();
+
+    File integrationTestsFile = new File(moduleDir, testExtension.integrationTestDir);
+    boolean hasIntegrationTests = testExtension.enableIntegrationTests && integrationTestsFile.isDirectory();
```

Comment:
this should be `target.getIntegrationTest().getSources().isEmpty()`

I think the scope might just return an empty configuration scope since the `integrationTest` configuration is not resolvable.

Reply:
This fails for java targets without the proper tasks. The way you described was my first attempt.
If we solve the way to automatically add the sourceSets and integrationTest task for all jvm targets on okbuck itself, this check is not needed anymore